pkg/kf/algorithms/algorithms.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package algorithms
+
+import (
+	"sort"
+)
+
+// Interface is the interface used by the algorithms.
+type Interface interface {
```

Comment:
This is nice, but are there going to be other places where it will be used? I'm fine with leaving this as is, just curious if you have any future use cases in mind

Reply:
My hope is that other reconcilers can get use out of this. I assume the need to delete, dedupe, etc is going to be necessary outside of routes.